entry.go

```diff
@@ -67,33 +68,42 @@ func (entry *Entry) WithFields(fields Fields) *Entry {
 	for k, v := range fields {
 		data[k] = v
 	}
-	return &Entry{Logger: entry.Logger, Data: data}
+	return &Entry{Logger: entry.Logger, Data: data, Level: entry.Level}
 }
 
 // This function is not declared with a pointer value because otherwise
 // race conditions will occur when using multiple goroutines
 func (entry Entry) log(level Level, msg string) {
-	entry.Time = time.Now()
-	entry.Level = level
-	entry.Message = msg
+	var cxtEntry *Entry = &entry
```

Comment:
AFAIK all the changes in this func are meant to accomodate the unnecessary copy, so unless I'm missing something, the func should be left intact.